### PR TITLE
Fail if process exits with non-zero status

### PR DESCRIPTION
**Pull request by danielhollas**
Follow up on https://github.com/aiidalab/aiidalab-widgets-base/pull/336, see discussion there.

(I tested this in my own app, but suppose it works here too)

### CODE REVIEW

**Review by unkcpz — APPROVED** [review]
@danielhollas thanks! Looks all good to me.

File: aiidalab_qe/steps.py [hunk 1]
```diff
@@ -927,9 +927,12 @@ def _update_state(self):
                 ProcessState.WAITING,
             ):
                 self.state = self.State.ACTIVE
-            elif process_state in (ProcessState.EXCEPTED, ProcessState.KILLED):
+            elif (
+                process_state in (ProcessState.EXCEPTED, ProcessState.KILLED)
+                or self.process.is_failed
+            ):
```
Comment (unkcpz) [inline aiidalab_qe/steps.py:933]: Are these all cases of `not self.process.is_finished_ok`? But this is more explicit and all good from my side.

Comment (danielhollas) [reply aiidalab_qe/steps.py:933]: Yes, as far as I know. Thanks!

**Comment by unkcpz** [comment]
Cheers!

**Pull request merged**